pkg/controller/customproperties/reconciler.go

```diff
@@ -0,0 +1,140 @@
+package customproperties
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/apis/dynatrace/v1alpha1"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	Suffix     = "-custom-properties"
+	DataKey    = "customProperties"
+	DataPath   = "custom.properties"
+	VolumeName = "custom-properties"
+	MountPath  = "/mnt/dynatrace/gateway/config"
+)
+
+type Reconciler struct {
+	client.Client
+	scheme                    *runtime.Scheme
+	log                       logr.Logger
+	customPropertiesSource    v1alpha1.DynaKubeValueSource
+	customPropertiesOwnerName string
+	instance                  *v1alpha1.DynaKube
+}
+
+func NewReconciler(clt client.Client, instance *v1alpha1.DynaKube, log logr.Logger, customPropertiesOwnerName string, customPropertiesSource v1alpha1.DynaKubeValueSource, scheme *runtime.Scheme) *Reconciler {
+	return &Reconciler{
+		Client:                    clt,
+		instance:                  instance,
+		scheme:                    scheme,
+		log:                       log,
+		customPropertiesSource:    customPropertiesSource,
+		customPropertiesOwnerName: customPropertiesOwnerName,
+	}
+}
+
+func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
param "request" seems to be never used - do we need it?

Reply:
replaced unused parameter with underscore